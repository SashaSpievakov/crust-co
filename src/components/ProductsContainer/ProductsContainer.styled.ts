import styled from 'styled-components';

interface SectionProps {
  noSearchResult: boolean;
}

const Section = styled.section<SectionProps>`
  display: grid;
  grid-template-columns: ${({ noSearchResult }) =>
    noSearchResult ? '1fr' : 'repeat(3, 1fr)'};
  gap: 40px 100px;
  justify-items: center;
  align-items: center;

  @media (max-width: 1150px) {
    grid-template-columns: ${({ noSearchResult }) =>
      noSearchResult ? '1fr' : 'repeat(2, 1fr)'};
  }

  @media (max-width: 800px) {
    gap: 40px 50px;
  }

  @media (max-width: 700px) {
    grid-template-columns: ${({ noSearchResult }) =>
      noSearchResult ? '1fr' : 'repeat(1, 1fr)'};
    gap: 0;
  }
`;

const SearchError = styled.h3`
  font-size: 35px;
  margin: 100px 0;

  @media (max-width: 400px) {
    font-size: 30px;
    padding: 0 10px;
  }
`;

export { Section, SearchError };
