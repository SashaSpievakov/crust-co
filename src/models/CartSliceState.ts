import { ICartItem } from './ICartItem';

export interface CartSliceState {
  totalPrice: number;
  itemsCount: number;
  items: ICartItem[];
}
