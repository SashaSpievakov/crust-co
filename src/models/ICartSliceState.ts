import { ICartItem } from './ICartItem';

export interface ICartSliceState {
  totalPrice: number;
  itemsCount: number;
  items: ICartItem[];
}
