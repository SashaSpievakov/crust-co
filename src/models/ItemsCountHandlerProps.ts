export interface ItemsCountHandlerProps {
  id: string;
  name: string;
  price: number;
  activeSize: number;
  activeType: number;
  typeNames: string[];
  isFullScreen?: boolean;
}
