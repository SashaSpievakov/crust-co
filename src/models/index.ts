export type { ICartItem } from './ICartItem';
export type { IPizzaItem } from './IPizzaItem';
