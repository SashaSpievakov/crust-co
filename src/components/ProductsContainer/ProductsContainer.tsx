import { useAppSelector } from '../../hooks/reduxHooks';
import { IPizzaItem } from '../../models/IPizzaItem';
import { selectSearchValue } from '../../store/slices/searchSlice';
import ItemCard from '../ItemCard/ItemCard';
import Skeleton from '../ItemCard/Skeleton';
import Container from './ProductsContainer.styled';

interface ProductsContainerProps {
  isLoading: boolean;
  items: IPizzaItem[];
}

const ProductsContainer = ({ isLoading, items }: ProductsContainerProps) => {
  const searchValue = useAppSelector(selectSearchValue);

  return (
    <Container>
      {isLoading
        ? [...new Array(9)].map((_, i) => <Skeleton key={i} />)
        : items
            .filter((item: IPizzaItem) =>
              item.name
                .toLowerCase()
                .includes(searchValue.trim().toLowerCase()),
            )
            .map((item: IPizzaItem) => <ItemCard key={item.id} {...item} />)}
    </Container>
  );
};
export default ProductsContainer;
