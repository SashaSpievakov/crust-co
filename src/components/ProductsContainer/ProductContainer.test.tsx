import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';

import ProductsContainer from './ProductsContainer';
import rendererWithAllProviders from '../../tests/helpers/rendererWithProviders';
import renderWithProviders from '../../tests/helpers/renderWithProviders';
import { IPizzaItem } from '../../models/IPizzaItem';

const ProductsContainerItems: IPizzaItem[] = [
  {
    id: '12',
    name: 'Vegetarian Pizza',
    description:
      'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Aliquid nisi aspernatur debitis quod suscipit, ',
    types: [0, 1],
    sizes: [12, 14, 16],
    price: 12,
    category: 2,
    rating: 8,
  },
  {
    id: '9',
    name: 'Meat Pizza',
    description:
      'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Ducimus, iste.',
    types: [0],
    sizes: [12, 16],
    price: 14,
    category: 1,
    rating: 15,
  },
  {
    id: '2',
    name: 'Pepperoni Pizza',
    description:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quam, cum recusandae! Quidem, quaerat',
    types: [0],
    sizes: [12, 14],
    price: 10,
    category: 4,
    rating: 5,
  },
];

describe('ProductsContainer Test', () => {
  test('renders the ProductsContainer component', () => {
    const snapshot = rendererWithAllProviders(
      <ProductsContainer isLoading={false} items={ProductsContainerItems} />,
    );
    expect(snapshot).toMatchSnapshot();
  });

  test('renders the ProductsContainer component skeletons', () => {
    renderWithProviders(
      <ProductsContainer isLoading items={ProductsContainerItems} />,
    );
    const skeletons = screen.getAllByRole('presentation');

    expect(skeletons[0]).toBeInTheDocument();
    expect(skeletons[8]).toBeInTheDocument();
  });

  test('renders components with a search value', () => {
    renderWithProviders(
      <ProductsContainer isLoading={false} items={ProductsContainerItems} />,
      false,
      '/',
      { searchValue: { value: '  Meat Pizza ' } },
    );
    const heading = screen.getByRole('heading');

    expect(heading).toHaveTextContent(/meat pizza/i);
  });
});
