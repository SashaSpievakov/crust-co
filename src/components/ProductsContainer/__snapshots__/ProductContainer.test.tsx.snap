// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsContainer Tests renders the ProductsContainer component 1`] = `
Object {
  "children": Array [
    <article
      className="sc-bqWxrE fptGBs"
    >
      <a
        href="/item/12"
        onClick={[Function]}
      >
        <picture>
          <source
            srcSet="./assets/img/pizza12.webp"
            type="image/webp"
          />
          <source
            srcSet="./assets/img/pizza12.png"
            type="image/png"
          />
          <img
            alt="Vegetarian Pizza pizza"
            className="sc-ksBlkl giKwGo"
            height="250"
            src="./assets/img/pizza12.png"
            width="250"
          />
        </picture>
        <h3
          className="sc-hBxehG cZCKzy"
        >
          Vegetarian Pizza
        </h3>
      </a>
      <div
        className="sc-csuSiG fpLAMx"
      >
        <ul>
          <li
            className="sc-eDWCr cQFBpw"
            onClick={[Function]}
          >
            traditional
          </li>
          <li
            className="sc-eDWCr jWyKRS"
            onClick={[Function]}
          >
            thin
          </li>
        </ul>
        <ul>
          <li
            className="sc-eDWCr cQFBpw"
            onClick={[Function]}
          >
            12
             inch
          </li>
          <li
            className="sc-eDWCr jWyKRS"
            onClick={[Function]}
          >
            14
             inch
          </li>
          <li
            className="sc-eDWCr jWyKRS"
            onClick={[Function]}
          >
            16
             inch
          </li>
        </ul>
      </div>
      <div
        className="sc-fnGiBr PZUmi"
      >
        <div
          className="sc-fEXmlR ljIWrJ"
        >
          12
          $
        </div>
        <button
          className="sc-bcXHqe sc-eDvSVe sc-gKPRtg fAfvZN fwWQTF gmBSoV"
          onClick={[Function]}
        >
          <svg
            className="sc-ipEyDJ fbIxLm"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"
            />
          </svg>
          <span>
            Add
          </span>
        </button>
      </div>
    </article>,
    <article
      className="sc-bqWxrE fptGBs"
    >
      <a
        href="/item/9"
        onClick={[Function]}
      >
        <picture>
          <source
            srcSet="./assets/img/pizza9.webp"
            type="image/webp"
          />
          <source
            srcSet="./assets/img/pizza9.png"
            type="image/png"
          />
          <img
            alt="Meat Pizza pizza"
            className="sc-ksBlkl giKwGo"
            height="250"
            src="./assets/img/pizza9.png"
            width="250"
          />
        </picture>
        <h3
          className="sc-hBxehG cZCKzy"
        >
          Meat Pizza
        </h3>
      </a>
      <div
        className="sc-csuSiG fpLAMx"
      >
        <ul>
          <li
            className="sc-eDWCr cQFBpw"
            onClick={[Function]}
          >
            traditional
          </li>
        </ul>
        <ul>
          <li
            className="sc-eDWCr cQFBpw"
            onClick={[Function]}
          >
            12
             inch
          </li>
          <li
            className="sc-eDWCr jWyKRS"
            onClick={[Function]}
          >
            16
             inch
          </li>
        </ul>
      </div>
      <div
        className="sc-fnGiBr PZUmi"
      >
        <div
          className="sc-fEXmlR ljIWrJ"
        >
          14
          $
        </div>
        <button
          className="sc-bcXHqe sc-eDvSVe sc-gKPRtg fAfvZN fwWQTF gmBSoV"
          onClick={[Function]}
        >
          <svg
            className="sc-ipEyDJ fbIxLm"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"
            />
          </svg>
          <span>
            Add
          </span>
        </button>
      </div>
    </article>,
    <article
      className="sc-bqWxrE fptGBs"
    >
      <a
        href="/item/2"
        onClick={[Function]}
      >
        <picture>
          <source
            srcSet="./assets/img/pizza2.webp"
            type="image/webp"
          />
          <source
            srcSet="./assets/img/pizza2.png"
            type="image/png"
          />
          <img
            alt="Pepperoni Pizza pizza"
            className="sc-ksBlkl giKwGo"
            height="250"
            src="./assets/img/pizza2.png"
            width="250"
          />
        </picture>
        <h3
          className="sc-hBxehG cZCKzy"
        >
          Pepperoni Pizza
        </h3>
      </a>
      <div
        className="sc-csuSiG fpLAMx"
      >
        <ul>
          <li
            className="sc-eDWCr cQFBpw"
            onClick={[Function]}
          >
            traditional
          </li>
        </ul>
        <ul>
          <li
            className="sc-eDWCr cQFBpw"
            onClick={[Function]}
          >
            12
             inch
          </li>
          <li
            className="sc-eDWCr jWyKRS"
            onClick={[Function]}
          >
            14
             inch
          </li>
        </ul>
      </div>
      <div
        className="sc-fnGiBr PZUmi"
      >
        <div
          className="sc-fEXmlR ljIWrJ"
        >
          10
          $
        </div>
        <button
          className="sc-bcXHqe sc-eDvSVe sc-gKPRtg fAfvZN fwWQTF gmBSoV"
          onClick={[Function]}
        >
          <svg
            className="sc-ipEyDJ fbIxLm"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"
            />
          </svg>
          <span>
            Add
          </span>
        </button>
      </div>
    </article>,
  ],
  "props": Object {
    "className": "sc-idXgbr cCVaxy",
  },
  "type": "div",
}
`;
